### PR TITLE
lua: Make building more generic (build for any arch)

### PR DESCRIPTION
**Pull request by Darchiv**
Default version changed from 5.3.5 to 5.3.6.

luaconf_local.h is required to be present in project's directory and
can be used to override configuration values in e.g. luaconf.h

Added port build options:
 - PORTS_LUA_CONFIG_DIR - path to a directory where luaconf_local.h is
   located. If not provided, the port's directory is used;
 - PORTS_LUA_VERSION - version to download (must be supported by
   respective patches);
 - PORTS_LUA_TESTS_VERSION - version of tests to download;
 - PORTS_LUA_BIN_CSTACK_SIZE - convenience option to set `-z stack-size`
   for lua binary built in this port;
 - PORTS_LUA_INSTALL_TESTS - download and install external Lua test
   scripts;
 - PORTS_LUA_SAFE - disable Lua features which may be used to crash
   the interpreter or perform side-effects in the operating system.

JIRA: NIL-595

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: imxrt117x-evk.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order): https://github.com/phoenix-rtos/phoenix-rtos-ports/pull/78
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: lua/build.sh [hunk 1]
```diff
@@ -2,34 +2,59 @@
...
+
+b_port_apply_patches "${PREFIX_PORT_SRC}" "${version}"
+
+if [ "${PORTS_LUA_RESTRAIN}" = "y" ]; then
+	b_port_apply_patches "${PREFIX_PORT_SRC}" "${version}/restrain"
 fi
 
+mycflags=(
+	${CFLAGS}
+	-DLUAI_MAXSTACK=${PORTS_LUA_STACK_SIZE:-2000}
```
Comment (mateuszkobak) [inline lua/build.sh:28]: `LUAI_MAXSTACK` passed to lualib should be compatible with the one that is given to applications (as it is used in one of the header files of Lua API). The value of  `PORTS_LUA_STACK_SIZE` will be used outside of ports too.

File: lua/build.sh [hunk 2]
```diff
@@ -2,31 +2,51 @@
...
-	cp "$PREFIX_PORT/Makefile" "$PREFIX_LUA_SRC/src/"
+
+b_port_apply_patches "${PREFIX_PORT_SRC}" "${version}"
+
+if [ "${PORTS_LUA_SAFE}" = "y" ]; then
+	b_port_apply_patches "${PREFIX_PORT_SRC}" "${version}/safe"
 fi
 
+myldflags=(
+	${LDFLAGS}
```
[the remark this answers is not in the document]
Comment (Darchiv) [reply lua/build.sh:27]: Split intended

File: lua/build.sh [hunk 3]
```diff
@@ -2,31 +2,51 @@
...
+	-Wl,-z,stack-size=${PORTS_LUA_BIN_CSTACK_SIZE:-4096}
```
[the remark this answers is not in the document]
Comment (Darchiv) [reply lua/build.sh:28]: Split intended

File: lua/build.sh [hunk 4]
```diff
@@ -2,31 +2,51 @@
...
-	tar zxf "$PREFIX_PORT/${LUA}.tar.gz" -C "$PREFIX_PORT_BUILD"
-	cp "$PREFIX_PORT/Makefile" "$PREFIX_LUA_SRC/src/"
+
+b_port_apply_patches "${PREFIX_PORT_SRC}" "${version}"
+
+if [ "${PORTS_LUA_SAFE}" = "y" ]; then
+	b_port_apply_patches "${PREFIX_PORT_SRC}" "${version}/safe"
 fi
 
+myldflags=(
```
[the remark this answers is not in the document]
Comment (Darchiv) [reply lua/build.sh:26]: Intended, commas are part of strings

[the remark this answers is not in the document]
Comment (nalajcie) [reply lua/build.sh:26]: put it in `"`?

File: lua/build.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (nalajcie) [reply lua/build.sh:27]: maaybe either silence the shellcheck warning `# shellcheck disable=SC2206` or use `read` to get variables into array

**Pull request merged**